doc/nrf/test_and_optimize.rst

```diff
@@ -1,17 +1,159 @@
 .. _test_and_optimize:
+.. _gs_testing:
+.. _serial_terminal_connect:
+.. _testing_rtt:
+.. _testing_rtt_connect:
+.. _putty:
+.. _testing_vscode:
 
 Testing and optimization
 ########################
 
-Each :ref:`applications` and :ref:`samples` in the |NCS| come with their own testing instructions.
-However, the |NCS| provides a common set of features and options for debugging and optimizing the application performance.
-This includes a multilevel logging system that can be enabled and configured independently for different modules, and support for writing unit tests using Unity and CMock.
+|application_sample_definition|
+
+The :ref:`applications` and :ref:`samples` in the |NCS| come with their own testing instructions.
+Follow these instructions to make sure that the application runs as expected.
```

Comment:
I noticed the include is causing a doubling up on links:
![image](https://github.com/nrfconnect/sdk-nrf/assets/83582261/24ebefeb-0868-4e48-9be4-15e8913489a7)

```suggestion
|application_sample_definition|

Each application in the |NCS| come with its own testing instructions.
Follow these instructions to make sure that the application runs as expected.
```